crates/sui-indexer/src/handlers/objects_snapshot_processor.rs

```diff
@@ -162,8 +193,36 @@ where
                             .latest_object_snapshot_sequence_number
                             .set(start_cp as i64);
                     }
+
+                    buffer_cp = self.buffer.lock().unwrap().startup_checkpoint;
                 }
             }
         }
+
+        // Now the cp in objects snapshot is greater than what's in the buffer, so we can start flushing
+        // the buffer to objects snapshot.
```

Comment:
Not sure if addressed but one potential issue might arise if snapshot exceeds buffer such that some object in snapshot may be a later version than what we'll end up upserting from buffer

Reply:
Yes it's indeed possible, although the buffer will contain later versions too and will overwrite with the correct version. To solve this, I can just add cp to each `TransactionObjectChanges` in the buffer and discard the ones that's before start_cp.

Reply:
Addressed in the latest commit.